subsys/sd/sdmmc.c

```diff
@@ -1248,13 +1248,11 @@ static int sdmmc_read(struct sd_card *card, uint8_t *rbuf,
 	}
 
 	/* Verify card is back in transfer state after read */
-	if (!card->host_props.is_spi) {
-		ret = sdmmc_wait_ready(card);
-		if (ret) {
-			LOG_ERR("Card did not return to ready state");
-			k_mutex_unlock(&card->lock);
-			return -ETIMEDOUT;
-		}
+	ret = sdmmc_wait_ready(card);
+	if (ret) {
+		LOG_ERR("Card did not return to ready state");
+		k_mutex_unlock(&card->lock);
```

Comment:
@danieldegrasse OT, mutex usage does not look sane, called from sdmmc_read_blocks() there would be second `k_mutex_unlock(&card->lock);`.